3rdparty/fastcv/src/fastcv_hal_imgproc.cpp

```diff
@@ -141,28 +221,88 @@ int fastcv_hal_sobel(
             CV_HAL_RETURN_NOT_IMPLEMENTED(cv::format("Border type:%s is not supported", borderToString(border_type)));
     }
 
-    int fcvFuncType = FCV_MAKETYPE(ksize,src_depth);
-
-    switch (fcvFuncType)
+    if (margin_left||margin_top||margin_top||margin_bottom)
     {
-        case FCV_MAKETYPE(3,CV_8U):
+        int src_height = height, src_width = width, start_height = 0, start_width = 0;
+
+        if(margin_left != 0)
         {
-            status = fcvFilterSobel3x3u8s16(src_data, width, height, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
-            break;
+            src_data    -= ksize/2;
+            src_width   += ksize/2;
+            start_width =  ksize/2;
         }
-        case FCV_MAKETYPE(5,CV_8U):
+
+        if(margin_top != 0)
         {
-            status = fcvFilterSobel5x5u8s16(src_data, width, height, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
-            break;
+            src_data     -= (ksize/2) * src_step;
+            src_height   += ksize/2;
+            start_height =  ksize/2;
         }
-        case FCV_MAKETYPE(7,CV_8U):
+
+        if(margin_right != 0)
         {
-            status = fcvFilterSobel7x7u8s16(src_data, width, height, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
-            break;
+            src_width += ksize/2;
         }
-        default:
-            CV_HAL_RETURN_NOT_IMPLEMENTED(cv::format("Ksize:%d, src_depth:%s, border type:%s is not supported",
-                ksize, cv::depthToString(src_depth), borderToString(border_type)));
+
+        if(margin_bottom != 0)
+        {
+            src_height += ksize/2;
+        }
+
+        std::vector<uchar> tmp(src_height * dst_step);
+        int16_t *dxBuffer = nullptr, *dyBuffer = nullptr;
+
+        if ((dx == 1) && (dy == 0))
+        {
+            dxBuffer = (int16_t*)tmp.data();
+            dyBuffer = NULL;
+        }
+        else if ((dx == 0) && (dy == 1))
+        {
+            dxBuffer = NULL;
+            dyBuffer = (int16_t*)tmp.data();
+        }
+
+        int fcvFuncType = FCV_MAKETYPE(ksize, src_depth);
+
+        switch (fcvFuncType)
+        {
+            case FCV_MAKETYPE(3,CV_8U):
+            {
+                status = fcvFilterSobel3x3u8s16(src_data, src_width, src_height, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
+                break;
+            }
+            case FCV_MAKETYPE(5,CV_8U):
+            {
+                status = fcvFilterSobel5x5u8s16(src_data, src_width, src_height, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
+                break;
+            }
+            case FCV_MAKETYPE(7,CV_8U):
+            {
+                status = fcvFilterSobel7x7u8s16(src_data, src_width, src_height, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
+                break;
+            }
+            default:
+                break;
```

Comment:
I propose to add at least CV_Assert here. Just for the case, if invoker code will be changed incorrectly in future.

Reply:
which param do you think we should add assert? or I can return CV_HAL_ERROR_NOT_IMPLEMENTED